generic_tracers/generic_COBALT_diagyaml2register_diag_body.py

```diff
@@ -0,0 +1,110 @@
+"""
+File: 
+    generic_COBALT_diagyaml2register_diag_body.py
+Author: 
+    Kelly Kearney
+Date: 
+    2024/03/27
+Description: 
+    This script builds the Fortran source code that makes up the primary body of the 
+    generic_COBALT_register_diag subroutine.  This routine allows for easy maintenance of the 
+    highly-repetitive and lengthy code.  The resulting generic_COBALT_register_diag_body.f90 is 
+    added to the primary generic_COBALT.f90 module via an include statement, and can be quickly 
+    regenerated when new diagnostics are added or the specific commands in question need to be 
+    modified.
+Usage:
+    python generic_COBALT_diagyaml2register_diag_body.py formatopt
+
+    formatopt is a string indicating the code format to use:
+        original:   The format in the original generic_COBALT.  A (temporary) vardesc-formatted 
+                    variable is created for each diag, then register_diag_field is called using 
+                    fields from that, using pass-by-position syntax for the first 6 inputs and 
+                    name/value for any additional (default if script called w/o input)
+        novardesc:  This format eliminates the creation of an intermediate vardesc variable 
+"""
+
+import yaml
+import sys
+
+# Parse and check input
+
+if len(sys.argv) < 1:
+    formatopt = "original"
+else:
+    formatopt = sys.argv[1]
+
+if not formatopt in ["original", "novardesc"]:
+    raise Exception("Unrecognized format option")
+
+# Read diagnostics from yaml file
+
+yamlfile = 'generic_COBALT_diag.yaml'
+with open(yamlfile, 'r') as f:
+    diag = yaml.safe_load(f)
+    
+    
+# List of primary parameter names.  These include the vardesc type fields and the inputs to
+# register_diag_field that were passed by position (rather than name/value) in the original
+# generic_COBALT code.  
+
+primary = ["name", "longname", "hor_grid", "z_grid", "t_grid", "units", "mem_size", 
```

Comment:
Since some of these appear to be always the same (hor_grid, t_grid, mem_size, missing_value), could we put defaults for those in the python file that can still be overwritten by the yaml if ever necessary? 

It also seems like technically we only need one of z_grid and axes, since 2d variables have z_grid=1 and axes=axes(1:2) and 3D variables have z_grid=L and axes=axes(1:3).

Reply:
There are some variables on the "1" z-axis (i.e. neither 2D nor 3D) where the axes variable can vary (either axes(1:2) or axesTi(1:1))... these are things like runoff and bottom fluxes.

But there is also a comment there indicating that the values of the axes variable is never really used, other than a length check, so there is possibly some simplification that could be done there.